### ISSUE TITLE
add racial traits

### ISSUE THREAD

**Issue by ramaiguy** [zombieFox/awesomeSheet#168]
Love this character sheet, thank you for your development. If you gave me a format, I'd be willing to do some gruntwork to get your sheet to be racially aware. Maybe we can start super simple, like just doing ability score modifiers for the races on d20PFSRD and then go from there. Thanks!

**Comment by zombieFox** [comment]
Hi, thanks for the feedback. Glad you are enjoying the app. 
I'm not sure I understand the issue you are describing. To help me understand better can you describe the problem or limitation you encountered while using the sheet? Or what problem a new proposed feature might solve?

**Comment by ramaiguy** [comment]
Hey zombieFox,
thanks for getting back to me. I was just using the sheet for my Gnome Slayer. There was no pull down menu for races, to have the racial bonuses smartly applied to the sheet. For a gnome, its +2 Constitution, +2 Charisma, and –2 Strength. It would be nice if the sheet was smart enough to input that bit of info automatically when you choose the race. So I was offering to input all of the racial bonuses into a useful database file of some sort to be applied to the sheet. I hope that helps! Thanks
-Ramai

**Comment by zombieFox** [comment]
Thanks for the details, that does help. 😄 
I've been investigating this idea of more automation -- there are pros and cons to providing automated Race stats. The baseline would be to set the racial stat modifiers on selection. eg:

Select "Dwarf" from a Race selection dropdown. Automatically add the following stats: 
```
  stats: {
    con: 2,
    wis: 2,
    cha: -1
  }
```
These stats would be locked unless a different race is selected from the Race selection dropdown.

The danger here is that the expectation has been set that making a selection from the Race selection dropdown populates the racial abilities of the chosen race. However there are more to racial abilities than Stat changes.  It's something more like this:
```
dwarf: {
  stats: {
    con: 2,
    wis: 2,
    cha: -1
  },
  size:	"medium",
  type: {
    string: "Humanoid",
    sub: "(dwarf)"
  },
  speed: 20,
  languages: ["Common", "Dwarven"],
  senses: ["Darkvision 60 ft."]
}
```

This feature represents a large task to develop. I will continue to investigate this feature request but for now I do not know if I can take on a task this size. I will happily accept help on project if you wish to develop it. I can support with direction and review any PRs.
Cheers.

**Comment by ramaiguy** [comment]
I agree about an expectation being there. I actually saw that awesomeSheet was aware of feats and traits, so I thought it would be neat if it was aware of more things since it's helpful. I am not a developer so I wouldn't be able to impliment this kind of racial awareness into awesomeSheet, but I would happily add all of the races listed on d20pfsrd into a format similar to what you've written above if that helps.

**Comment by DigiBear** [comment]
Just my own two cents I'd like to put in. I really love the AwesomeSheets openness letting you basically make everything yourself instead of being bound by the rules like e.g. herolab is. Though I do agree that adding a race feature and auto calculation can make it easier to quickly throw a character together.. the more you automate to more constrained the system becomes generally. 

Not saying of course that you shouldn't do it, just wish to give some input why I love AwesomeSheets so much and it would be a shame if you spend a lot of hours on something that might overall not improve much..

That said.. if you need help with content management for the race stuff, please let me know. I'd love to be capable of helping out.
